### PR TITLE
Refactor GitHub API integration on the Documentation project so it uses JSON file Feature/17436

### PR DESCRIPTION
**Pull request by accbjt**
Currently our Documentation project, the GitHub API integration uses github directory names to import markdown files into the project. We need a better manageable way to deal with this. We should have some sort of JSON file that will manage what is imported into the project.

### CODE REVIEW

File: packages/documentation/package.json [hunk 1]
```diff
@@ -31,7 +31,6 @@
     "react-dom": "^16.6.3",
     "react-helmet": "^5.2.0",
     "react-live": "^1.12.0",
-    "remark-mermaid": "^0.2.0",
```
Comment (accbjt) [inline packages/documentation/package.json]: we don't need this anymore.

**Comment by accbjt** [comment]
@jbalboni @rianfowler Let me know if this implementation looks okay.

File: packages/documentation/plugins/github-api-pages/gatsby-node.js [hunk 2]
```diff
@@ -1,37 +1,18 @@
-const path = require('path');
+const path  = require('path');
```
Comment (jbalboni) [inline packages/documentation/plugins/github-api-pages/gatsby-node.js]: Looks like this was accidental?

File: packages/documentation/src/pages/getting-started/common-tasks/creating-gatsby-plugins.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+---
```
Comment (jbalboni) [inline packages/documentation/src/pages/getting-started/common-tasks/creating-gatsby-plugins.mdx]: Can we add this page to the sidebar in common tasks?

File: packages/documentation/plugins/github-api-pages/pages/index.js [hunk 4]
```diff
@@ -0,0 +1 @@
+exports.workPractices = require('./work-practices.json');
```
Comment (jbalboni) [inline packages/documentation/plugins/github-api-pages/pages/index.js]: I think we could just have one JSON file, until we get to the point where that's too painful (which I think is unlikely).

File: packages/documentation/plugins/github-api-pages/gatsby-node.js (hunk 2 above)
Comment (accbjt) [reply packages/documentation/plugins/github-api-pages/gatsby-node.js]: okay

Comment (accbjt) [reply packages/documentation/plugins/github-api-pages/gatsby-node.js]: Yeah I think it was

**Pull request merged**